packages/@expo/metro-config/src/transform-worker/utils/require.ts

```diff
@@ -11,6 +11,10 @@ export async function tryRequireThenImport<TModule>(moduleId: string): Promise<T
     }
 
     if (requireError?.code === 'ERR_REQUIRE_ESM' && importESM) {
+      if (process.platform === 'win32') {
+        moduleId = moduleId.replace(/\\/g, '/');
+      }
+
       return (await importESM(moduleId)).default;
```

Comment:
I think this is a limitation in Node 18, but resolved in 20+. Try this instead:

```suggestion
      const moduleIdOrUrl = require('node:path').isAbsolute(moduleId)
        ? require('node:url').pathToFileURL(moduleId).href
        : moduleId;

      return (await importESM(moduleIdOrUrl)).default;
```

Granted, we'd still want the imports to be put at the top of the file here (but GitHub doesn't allow me to comment on that)